assets/js/googlesitekit/widgets/datastore/contexts.test.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * `core/widgets` data store: widget context tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { createTestRegistry } from '../../../../../tests/js/utils';
+import { CORE_WIDGETS } from './constants';
+import Null from '../../../components/Null';
+
+describe( 'core/widgets Widget context selectors', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	describe( 'isWidgetContextActive', () => {
+		it( 'requires a contextSlug', () => {
+			expect( () => {
+				registry.select( CORE_WIDGETS ).isWidgetContextActive();
+			} ).toThrow(
+				'contextSlug is required to check a widget context is active.'
+			);
+		} );
+
+		it( 'returns false if there are no areas registered for the context', () => {
+			expect(
+				registry
+					.select( CORE_WIDGETS )
+					.isWidgetContextActive( 'TestContext' )
+			).toEqual( false );
+		} );
+
+		describe( 'when there are areas registered for the context', () => {
```

Comment:
Same here. Please, move tests in this block to the upper level.

Reply:
The describe here does serve a functional purpose, as it enables the use of the beforeEach block.

Would you rather I extract the beforeEach setup into a function which can be invoked in each of the tests?

Reply:
That's done too :+1: